Character.cpp

```diff
@@ -86,32 +83,43 @@ int Character::takeDamage(int damage)
     return hitPoints;
 }
 
-
-#include <cassert>
 void Character::attackInternal(Character& other)
 {
-    if( other.hitPoints <= 0 )
+    if (other.hitPoints <= 0)
     {
         /*
         When you defeat another Character: 
             a) your stats are restored to their initial value if they are lower than it.
             b) your stats are boosted 10%
             c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
       */
-        assert(false);
-        std::cout << getName() << " defeated " << other.getName() << " and leveled up!" << std::endl;        
+        updateStat(armor, initialArmorLevel);
+        updateStat(hitPoints, initialHitPoints);
+        updateStat(attackDamage, initialAttackDamage);
+
+        std::cout << getName() << " defeated " << other.getName() << " and leveled up!" << std::endl;
     }
 }
 
+void Character::updateStat(int& currentValue, std::unique_ptr<int>& initialValue)
+{
+    if (currentValue < *initialValue)
+    {
+        currentValue = *initialValue;
+    }
+    currentValue *= 1.1f;
+    initialValue = std::make_unique<int>(currentValue);
+}
```

Comment:
Don't pass in the unique_ptr.   make the 2nd argument a `int&` and deference at the call-site.
don't make a new `initialValue`.  just update the existing one.
These steps will eliminate the 3 `*initialValue` in this function
- [ ] revise